### DISCUSSION TITLE
Crashloop on database connection

### DISCUSSION THREAD

**Discussion by elahmo** [plausible/analytics#4031] [Self-Hosted Support] [answered]
I am unable to get the app running:
```
2024-04-22T18:29:50.213826288Z app[web.1]:
2024-04-22T18:30:01.293321522Z app[web.1]: Loading plausible..
2024-04-22T18:30:01.302961122Z app[web.1]: Starting dependencies..
2024-04-22T18:30:01.428496152Z app[web.1]: Starting repos..
2024-04-22T18:30:01.572828501Z app[web.1]: create Plausible.Repo database if it doesn't exist
2024-04-22T18:30:01.719088610Z app[web.1]: ** (EXIT from #PID<0.95.0>) shutdown
```

`DATABASE_URL` and `CLICKHOUSE_DATABASE_URL` are set appropriately, and databases can be connected to. 

It appears that this [silently](https://github.com/plausible/analytics/blob/0621fc0a77cc50add00baddbdd59f378324da483/lib/plausible_release.ex#L95) fails for some reason. I have tried raising the LOG_LEVEL to debug to see whether something would pop up, without success.

Any idea what to check?

**Comment by ruslandoga** [comment]
👋 @elahmo 

Please try changing the compose command to just running the app to see if it provides more information on possible connection errors.

```diff
plausible:
    image: plausible/analytics:v2.0
-   command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh run"
+   command: sh -c "sleep 10 && /entrypoint.sh run"
```

**Reply by ruslandoga** [reply]
And also please check out https://github.com/plausible/analytics/discussions/3849 if you are using an external database provider. Sometimes they require the databases to be created through their UI.

**Reply by elahmo** [reply, marked as answer]
Thanks for the input! It appears the problem was with click house URL not being set up properly, in any case it would be helpful to see what was the reason for the crash as I had the URL set up but with different schema (`clickhouse://` instead of `http`) so it was a simple and, in hindsight, obvious fix, but I spent a few hours trying to figure out what was wrong :(
